integration/spark/src/main/scala/org/apache/spark/sql/execution/command/schema/CarbonAlterTableColRenameDataTypeChangeCommand.scala

```diff
@@ -195,18 +275,31 @@ private[sql] case class CarbonAlterTableColRenameDataTypeChangeCommand(
       var deletedColumnSchema: ColumnSchema = null
       var schemaEvolutionEntry: SchemaEvolutionEntry = null
       val columnSchemaList = tableInfo.fact_table.table_columns.asScala.filter(!_.isInvisible)
+      // to validate duplicate children columns
+      var UniqueColumnSet: mutable.Set[String] = mutable.Set()
```

Comment:
I feel this this place was okay to validate it. In validColumnsForRenaming() we only have a list of table columns. Also in validateComplexStructure() we traverse only the specific column that is being altered. 
Only columnSchemaList contains all columns (not only altered column but also other columns) as a flat structure,

Reply:
no, since we have one validation method already, please handle there itself as discussed last time, so that code will be clear and clean and any future changes will be easier